gcp_variant_transforms/testing/integration/run_preprocessor_tests.py

```diff
@@ -0,0 +1,178 @@
+
+_PIPELINE_NAME = 'gcp-variant-transforms-preprocessor-integration-test'
+_SCOPES = []
+_DEFAULT_ZONES = ['us-west1-b']
+_SCRIPT_PATH = '/opt/gcp_variant_transforms/bin/vcf_to_bq_preprocess'
+_TEST_FOLDER = 'gcp_variant_transforms/testing/integration/preprocessor_tests'
+
+
+class PreprocessorTestCase(run_tests_common.TestCase):
+  """Test case that holds information to run in pipelines API.
+
+  To define a new preprocessor_tests integration test case, create a json file
+  in gcp_variant_transforms/testing/integration/preprocessor_tests directory and
+  specify at least test_name, input_pattern, report_path and expected_contents
+  for the integration test.
+  """
+
+  def __init__(self,
+               context,
+               test_name,
+               input_pattern,
+               expected_contents,
+               report_path,
+               **kwargs):
+    # type: (TestContextManager, str, str, List[str], str, str) -> None
+    super(PreprocessorTestCase, self).__init__(test_name, context.project)
+    self._expected_contents = expected_contents
+    self._report_path = report_path
+    self._keep_reports = context.keep_reports
+    args = ['--input_pattern {}'.format(input_pattern),
+            '--report_path {}'.format(report_path),
+            '--project {}'.format(context.project),
+            '--staging_location {}'.format(context.staging_location),
+            '--temp_location {}'.format(context.temp_location),
+            '--job_name {}'.format(test_name)]
+    for k, v in kwargs.iteritems():
+      args.append('--{} {}'.format(k, v))
+
+    self._pipelines_api_request = {
+        'pipelineArgs': {
+            'projectId': context.project,
+            'logging': {'gcsPath': context.logging_location},
+            'serviceAccount': {'scopes': _SCOPES}
+        },
+        'ephemeralPipeline': {
+            'projectId': context.project,
+            'name': _PIPELINE_NAME,
+            'resources': {'zones': _DEFAULT_ZONES},
+            'docker': {
+                'imageName': context.image,
+                'cmd': ' '.join([_SCRIPT_PATH] + args)
+            }
+        }
+    }
+
+  def validate_result(self):
+    """Validates the results.
+
+    - Checks that the report is generated.
+    - Validates report's contents are the same as ``expected_contents``.
+    """
+    if not filesystems.FileSystems.exists(self._report_path):
+      raise run_tests_common.TestCaseFailure(
+          'Report is not generated in {}'.format(self._report_path))
+    with filesystems.FileSystems.open(self._report_path) as f:
+      for expected_content in self._expected_contents:
+        line = f.readline()
+        if expected_content != line:
+          raise run_tests_common.TestCaseFailure(
+              'Contents mismatch: expected {}, got {}'.format(
+                  expected_content, line))
+    if not self._keep_reports:
+      filesystems.FileSystems.delete(self._report_path)
+
+
+class TestContextManager(object):
```

Comment:
As I have commented below in `main()` I don't think that you need to follow the same pattern as the other test here. So I suggest that you remove `TestContextManager` completely from this module and instead user `args` directly in the constructor. This should make this test module simpler.